programs/vote/src/vote_state/mod.rs

```diff
@@ -809,8 +813,7 @@ pub fn withdraw<S: std::hash::BuildHasher>(
 ) -> Result<(), InstructionError> {
     let mut vote_account =
         instruction_context.try_borrow_instruction_account(vote_account_index)?;
-    let vote_state =
-        VoteStateHandler::deserialize_and_convert(&vote_account, VoteStateTargetVersion::V3)?;
+    let vote_state = VoteStateHandler::deserialize_and_convert(&vote_account, target_version)?;
```

Comment:
Also need to update the target version below in `deinitialize_vote_account_state`. I guess it needs tests too?